Update the What's New ctsm5.4 file (#3670)
ESCOMP/CTSM: tools/README.md

                           These scripts both help with creation of datasets using the
                           standard process as well as subsetting existing datasets and overwriting
                           some aspects for a specific case.

        modify_input_files Scripts to modify CTSM input files. Specifically modifying the surface
                           datasets and mesh files.

        contrib ---------- Miscellaneous tools for pre or post processing of CTSM.
                           Typically these are contributed by anyone who has something
                           they think might be helpful to the community. They may not
                           be as well tested or supported as other tools.

II. Notes on building/running for each of the above tools:

    mksurfdata_esmf has a cime configure and CMake based build using the following files:

        gen_mksurfdata_build ---- Build mksurfdata_esmf
        src/CMakeLists.txt ------ Tells CMake how to build the source code
        Makefile ---------------- GNU makefile to link the program together
        cmake ------------------- CMake macros for finding libraries

>     mkmapgrids, and site_and_regional only contain scripts so don't have the above build files.

Reviewer: ```suggestion
    mkmapgrids and site_and_regional only contain scripts that do not need build files.
```
